### PR TITLE
Update test-various to Ubuntu 20.04

### PR DESCRIPTION
**Pull request by nikic**
The test command-setgroups.rs is adjusted to skip on musl, where `sysconf(_SC_NGROUPS_MAX)` always returns a dummy value of 32, even though the actual value is 65536. I'm not sure why this only became a problem now, as the information I found indicates that this value changed in kernel version 2.6.4, which is ages ago.

I'm a bit unsure whether this one will go through, because I locally also saw a failure in std-backtrace.rs which went away on subsequent runs, and also had port assignment failures, but I think those might be on my side. I'm kind of curious how the code in https://github.com/rust-lang/rust/blob/b122908617436af187252572ed5db96850551380/library/std/src/net/test.rs#L43-L56 is supposed to work, as the directory names it checks don't seem to appear anywhere else.

r? @Mark-Simulacrum

### CODE REVIEW

File: src/ci/docker/host-x86_64/test-various/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-FROM ubuntu:18.04
+FROM ubuntu:20.04
 
-RUN apt-get update && apt-get install -y --no-install-recommends \
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
```
Comment (pickfire) [inline src/ci/docker/host-x86_64/test-various/Dockerfile:3]: Why is this necessary when we are already using `-y`?

Comment (nikic) [reply src/ci/docker/host-x86_64/test-various/Dockerfile:3]: Avoids prompts while configuring packages -- specifically necessary for tzdata.
